shell/platform/android/io/flutter/plugin/platform/VirtualDisplayController.java

```diff
@@ -145,15 +146,24 @@ public void run() {
           public void onViewDetachedFromWindow(View v) {}
         });
 
-    presentation =
+    // Initialize a new presentation. We don't want to cancel
+    // the old one before we can properly show the new one to
+    // take over where it left off. Cancelling before the show()
+    // of the new presentation causes the contents to stop, eg,
```

Comment:
I'm still confused by this, the view is detached from the presentation at the point we are cancelling it, not sure why would it cause a video to stop or why would showing a different presentation with that view will prevent the video from stopping?

Do you happen to which methods are invoked on the view in which order in the two scenarios? (the one where it stops and the one where it continue playing?)

Reply:
So I don't think I know enough about the inner workings of this to write a fully descriptive comment here. Most of this is from what I have gleaned via experimentation getting this to work. My comments here are mainly trying to explain the ordering that needs to happen, but I'm not 100% clear on specifically what in the implementations makes this necessary other that other orderings do not work. I don't want to pick out anything too specific as it may pin the cause incorrectly.

Any suggestions?